recipes/knowledge_distillation_single_device.py

```diff
@@ -582,13 +584,17 @@ def save_checkpoint(self, epoch: int) -> None:
         intermediate_checkpoint = epoch + 1 < self.total_epochs
         # if training is in-progress, checkpoint the optimizer state as well
         if intermediate_checkpoint:
+            dataloader_sd = self._dataloader.state_dict()
```

Comment:
Can you actually switch to using the hack from #2431 to advance the iterator? It's (slightly) cleaner than modifying the state dict directly.

Reply:
resolved in https://github.com/pytorch/torchtune/pull/2442/commits/6d4495aedb1c39d1ae00b2f6c8b2afb7664f286d

Reply:
You still need to advance the iterator after loading with `list(dataloader)`. See #2431

Reply:
[cc6988c](https://github.com/pytorch/torchtune/pull/2442/commits/cc6988cc32c9060bd4fecea0770c3a91be6b73a6) is not right, you'll just want to advance the iterator after loading the state dict into the dataloader as I mentioned above.

Reply:
ptl @ [32c84c5](https://github.com/pytorch/torchtune/pull/2442/commits/32c84c560882aae017cff0c946c7e7850e19aa61)